MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -319,14 +322,16 @@ - (void)onProcessingLog:(id<MSLog>)log withPriority:(MSPriority)priority {
   @synchronized(self) {
     NSData *serializedLog = [NSKeyedArchiver archivedDataWithRootObject:log];
     if (serializedLog && (serializedLog.length > 0)) {
-      if (self.bufferIndex > (ms_crashes_log_buffer_size - 1)) {
-        self.bufferIndex = 0;
+      if ([[self.bufferIndex objectForKey:@(priority)] integerValue] > (ms_crashes_log_buffer_size - 1)) {
+        // Reset the counter to 0 for a priority.
+        [self.bufferIndex setObject:@0 forKey:@(priority)];
       }
-      msCrashesLogBuffer[self.bufferIndex].buffer =
+      NSInteger index = [[self.bufferIndex objectForKey:@(priority)] integerValue];
+      msCrashesLogBuffer[priority][index].buffer =
           std::string(&reinterpret_cast<const char *>(serializedLog.bytes)[0],
                       &reinterpret_cast<const char *>(serializedLog.bytes)[serializedLog.length]);
-
-      self.bufferIndex += 1;
+      index += 1;
+      [self.bufferIndex setObject:@(index) forKey:@(priority)];
```

Comment:
Yeah, that's definitely one where clarity wins over brevity, as a rule. I have a tendency to write extremely dense code because I can read extremely dense code due to ~brain damage~ lots of practice 😅